addons/account/models/chart_template.py

```diff
@@ -606,6 +610,158 @@ def _setup_utility_bank_accounts(self, template_code, company, template_data):
+                    tax_group[field] = existing_accounts[account_template_xml_id]
+
+            return tax_groups
+
+        def get_adapted_tax_data(chart_template_code):
+            taxes = self._get_chart_template_data(chart_template_code)['account.tax']
+
+            for tax_template in taxes.values():
+                tax_template['country_id'] = country.id  # This is required because the country isn't provided directly by the template
+
+                if tax_template.get('tax_group_id'):
+                    tax_template['tax_group_id'] = f"{chart_template_code}_{tax_template['tax_group_id']}"
+
+                # Create and set accounts corresponding to the current chart template
+                for _command, _id, rep_line in tax_template.get('repartition_line_ids', []):
+                    if 'account_id' in rep_line and rep_line['repartition_type'] == 'tax':
+                        rep_line['account_id'] = corresponding_tax_repartition_line_account(tax_template['type_tax_use'], rep_line)
+
+            local_cash_basis_tax = self.env["account.tax"].search([
+                ('country_id', '=', company.account_fiscal_country_id.id),
+                ('cash_basis_transition_account_id', '!=', False)
+            ], limit=1)
+            for tax_template in taxes.values():
+                if not tax_template.get('cash_basis_transition_account_id'):
+                    continue
+                account_xml_id = tax_template['cash_basis_transition_account_id']
+
+                if account_xml_id in existing_accounts:
+                    tax_template['cash_basis_transition_account_id'] = existing_accounts[account_xml_id]
+                    continue
+
+                if local_cash_basis_tax:
+                    existing_accounts[account_xml_id] = create_foreign_tax_account(local_cash_basis_tax.cash_basis_transition_account_id, _("Cash basis transition account")).id
+                    tax_template['cash_basis_transition_account_id'] = existing_accounts[account_xml_id]
+                    continue
+
+                account_id = [rep_line['account_id'] for _command, _id, rep_line in tax_template['repartition_line_ids'] if rep_line.get('account_id')]
+                if account_id:
+                    local_account = self.env['account.account'].browse(account_id[0])
+                    existing_accounts[account_xml_id] = create_foreign_tax_account(local_account, _("Cash basis transition account")).id
+                    tax_template['cash_basis_transition_account_id'] = existing_accounts[account_xml_id]
+                    continue
+
+                tax_template['cash_basis_transition_account_id'] = False
+
+            return taxes
+
+        chart_template_code = self._guess_chart_template(country=country)
+        data = {
+            'account.tax.group': get_adapted_tax_group_data(chart_template_code),
+            'account.tax': get_adapted_tax_data(chart_template_code),
```

Comment:
I feel like I could just inline those 2 functions now.
It sounds nice in the beginning but doesn't that much anymore

Reply:
I mean, instead of
```
def big_funct(blabla):
    def tax_group_func(group_blabla):
        tax_groups_data = tax_group_stuff
        ...
        return tax_groups_data

    def tax_func(tax_blabla):
        tax_data = tax_stuff
        ....
        return tax_data
    
    return {'tax_group': tax_group_func(blabla), 'tax': tax_func(blabla)}
```

I could do:
```py
def big_funct(blabla):

    tax_groups_data = tax_group_stuff
    ... # modifying tax_group_data

    tax_data = tax_stuff
    .... # modifying tax_data
    
    return {'tax_group': tax_groups_data, 'tax': tax_data}
```

Reply:
But nevermind, if it's ok for you, let's keep it that way.
that's less work for now